libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,43 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (_unaryOperation.annotation().userDefinedFunction.set())
+	{
+		FunctionDefinition const* function = *_unaryOperation.annotation().userDefinedFunction;
+		solAssert(
+			function->isFree() || function->libraryFunction(),
+			"Only file-level functions and library functions can be bound to a user type operator."
+		);
+
+		FunctionType const* functionType = dynamic_cast<FunctionType const*>(
+			function->libraryFunction() ? function->typeViaContractName() : function->type()
+		);
+		solAssert(functionType);
```

Comment:
I've seen this bit repeated a few times. Maybe we could add a `userDefinedFunctionType()` helper for it in `UnaryOperation` and `BinaryOperation` in `AST.h`?

Reply:
Change not pushed.